#!/usr/bin/env python
PACKAGE = "solabot_rpi"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name              Type      Level    Description                Default   Min   Max
gen.add("linear_x_gain", double_t, 0, "cmd_vel x gain",    1.0, 0,   10)
gen.add("linear_y_gain", double_t, 0, "cmd_vel y gain",    1.0, 0,   10)
gen.add("angular_z_gain", double_t, 0, "cmd_vel angular_z gain",    1.0, 0,   10)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
# The third parameter should be equal to the cfg file name, without extension. Otherwise the libraries will be generated in every build, forcing a recompilation of the nodes which use them.
exit(gen.generate(PACKAGE, "decode_vel_serial_out_node", "decode_vel_serial_out"))
